File: LeetCode/560.py
class Solution(object):
    def subarraySum(self, nums, k):
        s = 0
        d = {0: 1}
        count = 0
        for j, num in enumerate(nums):
            s += nums[j]
            if d.get(s - k):
                count += d[s - k]
            if d.get(s):
                d[s] += 1
            else:
                d[s] = 1
        return count

File: LeetCode/560.cc
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int prefix{0};
        unordered_map<int,int> m;
        int ans{0};
        for(auto&& num : nums) {
            m[prefix]++;
            prefix += num;
            if(auto it = m.find(prefix-k); it != m.end())
                ans += it->second;
        }
        return ans;
    }
};